lablup/backend.ai-console
# feat: add support for signup user email verification

This PR resolves #536.

* New component: `backend-ai-email-verification-view`.
* Add route `verify-email` for email verification.
  - Visiting this url will try to verify token, not login.
  - Token should be delivered as a query string by `verification_code`.
    - If verified, success dialog is popup and redirects to root page.
    - If not verified, fail dialog is popped-up with re-send verification email form.
  - If token is not supplied, fail dialog is popped-up with re-send verification email form.
* New `Cloud` class in the JavaScript SDK.
  - Add `verify_email` and `send_verification_email` in the `Cloud` API.

## How to test signup with email verification feature
* `cloud-beta-plugin` should be installed (by pip) in the manager.
* Signup with test account (with invitation token). You will see a dialog which states that an email is sent. Check your mailbox to discover delivered email.
  ![image](https://user-images.githubusercontent.com/7539358/82153463-8a431f00-98a2-11ea-998c-f7b5d1f73d9c.png)
* Click VERIFY link. You should see email verified dialog. Or, you can see fail dialog if token is expired, or some error occurs.
  ![image](https://user-images.githubusercontent.com/7539358/82153396-2882b500-98a2-11ea-9e51-ee9342baca1d.png)
* In case of failure, user can resend a verification email again in the fail dialog.
